plugins/modules/query_graphql.py

```diff
@@ -0,0 +1,153 @@
+        required: False
+        default: True
+        type: bool
+    variables:
+        description:
+            - Dictionary of keys/values to pass into the GraphQL query, see [pynautobot GraphQL documentation](https://pynautobot.readthedocs.io/en/latest/advanced/graphql.html) for more details
+        required: False
+        type: dict
+"""
+
+EXAMPLES = """
+  # Make API Query without variables
+  - name: SET FACT OF STRING
+    set_fact:
+      query_string: |
+        query {
+          sites {
+            id
+            name
+            region {
+              name
+            }
+          }
+        }
+
+  # Make query to GraphQL Endpoint
+  - name: Obtain list of sites from Nautobot
+    networktocode.nautobot.query_graphql:
+      url: http://nautobot.local
+      token: thisIsMyToken
+      query: "{{ query_string }}"
+
+
+  # Example with variables
+  - name: SET FACTS TO SEND TO GRAPHQL ENDPOINT
+    set_fact:
+      variables:
+        site_name: den
+      query_string: |
+        query ($site_name:String!) {
+            sites (name: $site_name) {
+            id
+            name
+            region {
+                name
+            }
+            }
+        }
+
+  # Get Response with variables
+  - name: Obtain list of devices at site in variables from Nautobot
+    networktocode.nautobot.query_graphql:
+      url: http://nautobot.local
+      token: thisIsMyToken
+      query: "{{ query_string }}"
+      variables: "{{ variables }}"
+"""
+
+RETURN = """
+  data:
+    description:
+      - Data result from the GraphQL endpoint
+    type: dict
+    returned: success
+  url:
+    description:
+      - Nautobot URL that was supplied for troubleshooting
+    returned: success
+    type: str
+  query:
+    description:
+      - Query string that was sent to Nautobot
+    returned: success
+    type: str
+  variables:
+    description:
+      - Variables passed in
+    returned: success
+    type: dict
+"""
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """Main definition of Action Plugin for query_graphql."""
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(changed=False, original_message="", message="")
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=dict(
+            query=dict(required=True, type="str"),
+            token=dict(required=False, type="str", no_log=True, default=None),
+            url=dict(required=False, type="str", default=None),
```

Comment:
Should `url` be required?

Reply:
If we make it required, I don't think we get the option of loading it through an environment variable. I'll setup some time to walk through this and collaborate on the PR.